yt_dlp/extractor/telemb.py

```diff
@@ -1,75 +1,164 @@
+            'id': '36711',
+            'ext': 'mp4',
+            'title': 'Le CMA de Jemappes fêtera ses 20 ans ce week-end',
+            'creators': ['Loélia Chais'],
+            'description': 'md5:d41d8cd98f00b204e9800998ecf8427e',
+            'display_id': 'le-cma-de-jemappes-fetera-ses-20-ans-ce-week-end',
+            'duration': 316.08,
+            'location': 'Mons',
+            'release_date': '20241128',
+            'release_timestamp': 1732787226,
+            'thumbnail': r're:https?://www\.telemb\.be/cdn.+\.(?:jpe?g|png)',
+            'timestamp': 1732727988,
+            'upload_date': '20241127',
+        },
+    }]
 
     def _real_extract(self, url):
-        mobj = self._match_valid_url(url)
-        video_id = mobj.group('id')
-        display_id = mobj.group('display_id')
+        video_id = self._match_id(url)
+        display_id = self._match_valid_url(url).group('alt_id').split('/')[-1]
+        webpage = self._download_webpage(url, video_id)
 
-        webpage = self._download_webpage(url, display_id)
+        player_info = traverse_obj(webpage, (
+            {find_element(cls='freecaster-player', html=True)}, {extract_attributes}, {
+                'id': ('data-video-id', {str_or_none}),
+                'thumbnail': ('data-poster', {lambda x: self._proto_relative_url(x)}, {url_or_none}),
+            },
+        ))
+        video_info = self._download_json(
+            f'https://tvlocales-player-v12.freecaster.com/embed/{player_info.pop("id")}.json', video_id,
+        ).get('video')
 
+        qualities = {
+            '3': (640, 360),
+            '5': (960, 540),
+            '9': (1280, 720),
+            '11': (1920, 1080),
+        }
         formats = []
-        for video_url in re.findall(r'file\s*:\s*"([^"]+)"', webpage):
-            fmt = {
-                'url': video_url,
-                'format_id': video_url.split(':')[0],
-            }
-            rtmp = re.search(r'^(?P<url>rtmp://[^/]+/(?P<app>.+))/(?P<playpath>mp4:.+)$', video_url)
-            if rtmp:
-                fmt.update({
-                    'play_path': rtmp.group('playpath'),
-                    'app': rtmp.group('app'),
-                    'player_url': 'http://p.jwpcdn.com/6/10/jwplayer.flash.swf',
-                    'page_url': 'http://www.telemb.be',
-                    'preference': -10,
-                })
-            formats.append(fmt)
+        for src in traverse_obj(video_info, ('src', lambda _, v: v['src'])):
+            src_url = src['src']
+            ext = mimetype2ext(src.get('type'))
 
-        title = remove_start(self._og_search_title(webpage), 'TéléMB : ')
-        description = self._html_search_regex(
-            r'<meta property="og:description" content="(.+?)" />',
-            webpage, 'description', fatal=False)
-        thumbnail = self._og_search_thumbnail(webpage)
+            if ext == 'mp4':
+                quality = src_url.rpartition('_')[2].removesuffix('.mp4')
```

Comment:
Or if not requiring `url_or_none()` in general as above, this case should have it.
```suggestion
                if not src_url := url_or_none(src_url)
                    continue
                quality = src_url.rpartition('_')[2].removesuffix('.mp4')
```